File: LeetCode/LeetCode Practice Problems/1557- Minimum Number of Vertices to Reach All Nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> a;
        vector<int> seen(n,0);
        
        for(auto i: edges) {
            seen[i[1]] = 1;
        }
        
        for(int i=0;i<n;i++) {
            if(seen[i] == 0)
                a.push_back(i);
        }
        
        return a;
    }
};

File: LeetCode/LeetCode Practice Problems/1557-Minimum Number of Vertices to Reach All Nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        boolean vis[] = new boolean[n];
        
        for(List<Integer> edge: edges) {
            int u = edge.get(0);
            int v = edge.get(1);
            
            vis[v] = true;
        }
        
        List<Integer> ans = new ArrayList();
        
        for(int i=0;i<n;i++) {
            if(!vis[i]) {
                ans.add(i);
            }
        }
        
        return ans;
    }
}
